pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/ModuleVersionUpload.java

```diff
@@ -103,13 +121,19 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
                 draftSrc.jcrContent().getOrCreate()
                         .mimeType().set("text/x-asciidoc");
 
-                // TODO Html can no longer be generated on upload since there might be too many variants
-                // TODO This will need to be re-thought since metadata now lies on the variant
+                resolver.commit();
+
+                Map<String, Object> context = asciidoctorService.buildContextFromRequest(request);
+                asciidoctorService.getModuleHtml(module, localeObj, module.getWorkspace().getCanonicalVariantName(),
+                        true, context, true);
+
+                Metadata metadata = moduleLocale.variants().getOrCreate().variant(moduleLocale.getWorkspace().getCanonicalVariantName()).getOrCreate().draft().getOrCreate().metadata().getOrCreate();
```

Comment:
lets break this up into multiple lines, both for readability and in case we are hit by NPEs